org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/CompilationUnitDocumentProvider.java

```diff
@@ -434,83 +435,86 @@ public boolean isQuickFixable() {
 	}
 
 	/**
-	 * Internal structure for mapping positions to some value.
+	 * Internal structure for mapping positions to JavaMarkerAnnotations.
 	 * The reason for this specific structure is that positions can
 	 * change over time. Thus a lookup is based on value and not
 	 * on hash value.
 	 */
-	protected static class ReverseMap {
+	protected static class ReverseJavaMarkerAnnotationsMap {
```

Comment:
Some hints: 

1. The visibility of `annotations` should be >= than the visibility of this class, otherwise you may end up knowing about the class but can't really use it properly (in subclasses or classes in the same package)
2. Can this class be `private` instead of `protected`?

Reply:
From my point of view, the class should be `private`. I just did not change its prior visibility, but actually I would prefer your suggestion to have it `private`.

Reply:
I have adapted the visibility in 576aa64.